app/services/creator_analytics/churn.rb

```diff
@@ -0,0 +1,273 @@
+# frozen_string_literal: true
+
+class CreatorAnalytics::Churn
+  include ActiveModel::Validations
+
+  validates :end_date, comparison: { greater_than: :start_date }
```

Comment:
Should be `greater_than_or_equal_to` instead of `greater_than` to allow same-day analysis.